test/StreamJsonRpc.Tests/ActivityTracingStrategyTests.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using StreamJsonRpc;
+using StreamJsonRpc.Protocol;
+using Xunit;
+using Xunit.Abstractions;
+
+public class ActivityTracingStrategyTests : TestBase
+{
+    private readonly ActivityTracingStrategy strategy = new ActivityTracingStrategy();
+    private readonly JsonRpcRequest request = new JsonRpcRequest { Method = "test" };
+
+    public ActivityTracingStrategyTests(ITestOutputHelper logger)
+        : base(logger)
+    {
+    }
+
+    [Fact]
+    public void Outbound_NoContextualActivity()
+    {
+        this.strategy.ApplyOutboundActivity(this.request);
+        Assert.Null(this.request.TraceParent);
+        Assert.Null(this.request.TraceState);
+    }
+
+    [Fact]
+    public void Outbound_WithContextualActivity_WrongFormat_IsIgnored()
+    {
+        Activity.Current = new Activity("test").SetIdFormat(ActivityIdFormat.Hierarchical).Start();
+        try
+        {
+            this.strategy.ApplyOutboundActivity(this.request);
+            Assert.Null(this.request.TraceParent);
+            Assert.Null(this.request.TraceState);
+        }
+        finally
+        {
+            Activity.Current.Stop();
+        }
+    }
+
+    [Theory]
+    [InlineData(null)]
+    [InlineData("k=v")]
```

Comment:
nit: It probably makes sense to add a `[InlineData(k=v;k2=v2)]` to all of the tracestate tests here.

Reply:
Good idea. I added it here and to `CorrelationManagerTracingStrategyTests` as well. I used `,` as the delimiter since the spec uses that rather than `;`

But neither of these test *serialization* of the value. And since each formatter is responsible for serialization, we should test both of those. I'll add tests to `JsonRpcTests` to add `tracestate` and verify it's intact on the other side.

Reply:
...and it turns out I had a bug in serializing tracestate in messagepack that the new tests found. So thank you very much for your comment.